keps/sig-scheduling/3838-pod-mutable-scheduling-directives/README.md

```diff
@@ -0,0 +1,950 @@
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+- Allow mutating scheduling directives of pods unconditionally.
+- Restrict updating tolerations on scheduling gates. We already allow adding tolerations 
+  unconditionally, and so restricting that to gated pods is a backward incompatible change.
+- Allow mutating pod affinity constraints. It is not clear how this can be done without
+  potentially violating a policy check that we previously done at pod CREATE.
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+The proposal is to relax update validation of scheduling directives, specifically node affinity and selector, of Pods that
+are blocked on a scheduling readiness gate (i.e., not yet considered for scheduling by kube-scheduler), specifically. Note
+that adding tolerations and mutating labels and annotations are already allowed.
+
+This has no impact on the applications controllers. The applications controller generally has no dependency
+on those scheduling directives expressed in the application's pod template.
+
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+I want to build a controller that implements workload queueing and influences when and where a 
+workload should run. The workload controller creates the pods a workload instance, and to control 
+when and where the workload can run, I have a webhook that injects a scheduling readiness gate, 
+and a controller that removes the gate when capacity becomes available.
+
+At the time of creating the workload, we may not know on which part of the cluster (.e.g., a zone or 
+a VM type) a workload should run. To control where a workload should run, the controller can update 
+node affinity of the workload's pods; by doing so, the queue controller is able to influence the 
+scheduler and autoscaler decisions.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+- New calls from a queue controller to the API server to update affinity or tolerations. 
```

Comment:
Are you trying to say "selector" instead of "tolerations"?

Reply:
yes, updated.